reframe/utility/jsonext.py

```diff
@@ -17,10 +18,31 @@ def __rfm_json_encode__(self):
             '__rfm_file__': inspect.getfile(type(self))
         }
         ret.update(self.__dict__)
-        return ret
+        encoded_ret = encode_dict(ret, recursive=True)
+        _ret = encoded_ret if encoded_ret else ret
+        return _ret
 
 
-def encode(obj):
+def encode_dict(obj, *, recursive=False):
```

Comment:
There could be a need for an equivalent `decode_dict` to convert tuple like strings to tuples when loading the json.